WooCommerce/Classes/ViewRelated/Orders/Order Details/Issue Refunds/UseCases/RefundCreationUseCase.swift

```diff
@@ -0,0 +1,89 @@
+
+    /// Whether payment refund on the payment gateway should be attempted or not.
+    ///
+    let automaticallyRefundsPayment: Bool
+
+    /// Line items to be refunded, currently only order items are suported
+    ///
+    let items: [RefundableOrderItem]
+
+    /// Currency formatted needed for decimal calculations
+    ///
+    let currencyFormatter: CurrencyFormatter
+
+    /// Creates a `Refund` object ready to be used  on `RefundStore.createRefund` action
+    ///
+    func createRefundObject() -> Refund {
+        return Refund(refundID: .min,
+                      orderID: .min,
+                      siteID: .min,
+                      dateCreated: .distantPast,
+                      amount: amount,
+                      reason: reason ?? "",
+                      refundedByUserID: .min,
+                      isAutomated: nil,
+                      createAutomated: automaticallyRefundsPayment,
+                      items: createRefundItems())
+    }
+
+    /// Retuns an array of `OrderItemRefund` based on the provided refundable items
+    ///
+    private func createRefundItems() -> [OrderItemRefund] {
+        items.map { refundable -> OrderItemRefund in
+            OrderItemRefund(itemID: refundable.item.itemID,
+                            name: "",
+                            productID: .min,
+                            variationID: .min,
+                            quantity: Decimal(refundable.quantity),
+                            price: .zero,
+                            sku: nil,
+                            subtotal: "",
+                            subtotalTax: "",
+                            taxClass: "",
+                            taxes: createTaxes(from: refundable),
+                            total: calculateTotal(of: refundable),
+                            totalTax: "")
+        }
+    }
+
+    /// Creates an array of `OrderItemTaxRefund` from the tax lines in the provided `RefundableOrderItem`
+    ///
+    private func createTaxes(from refundable: RefundableOrderItem) -> [OrderItemTaxRefund] {
+        refundable.item.taxes.map { taxLine -> OrderItemTaxRefund in
+            OrderItemTaxRefund(taxID: taxLine.taxID,
+                               subtotal: "",
+                               total: calculateTax(of: taxLine, purchasedQuantity: refundable.item.quantity, refundQuantity: refundable.decimalQuantity))
+        }
+    }
+
+    /// Calculates the refundable tax from a tax line by diving its total tax value by the purchased quantity and mutiplying it by the refunded quantity.
+    ///
+    private func calculateTax(of taxLine: OrderItemTax, purchasedQuantity: Decimal, refundQuantity: Decimal) -> String {
+        let totalTax = currencyFormatter.convertToDecimal(from: taxLine.total) ?? 0
+        let itemTax = (totalTax as Decimal) / purchasedQuantity
+        let refundableTax = itemTax * refundQuantity
+        return currencyFormatter.localize(refundableTax) ?? "\(refundableTax)"
```

Comment:
just curious, does the API expect formatted amount for a refund with the site currency settings? I just thought that the formatting is usually for UI display. Also, since we only fetch site settings on app launch, there's a slight chance that currency settings are outdated

Reply:
From what I could see this method formats the amount without the currency and uses a `.` separator. 
It felt more secure than using only `"\(amount)"`

In any case I'll add a task to test this with different currencies and variations!